src/middlewared/middlewared/plugins/dyndns.py

```diff
@@ -55,6 +55,17 @@ class Config:
             'ipv6tb@he.net': 'he.net'
         }
 
+    @private
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+        if data.get('provider') not in await self.provider_choices():
```

Comment:
Thank you, @william-gr, my bad, the provider `custom` also should be accepted; also, why `.get`?